api/v1alpha1/etcdpeer_types.go

```diff
@@ -81,6 +143,88 @@ type EtcdPeerList struct {
 	Items           []EtcdPeer `json:"items"`
 }
 
+var _ webhook.Validator = &EtcdPeer{}
+
+// ValidateCreate validates that all required fields are present and valid.
+func (o *EtcdPeer) ValidateCreate() error {
```

Comment:
Instead of defining these on the actual EtcdPeer type, can we instead create an EtcdPeerValidator struct or something that exposes ValidateCreate/ValidateDelete etc.? Again, doesn't need to be in this PR, but thinking longer term 😄

Reply:
I'm not so sure about that. I'll postpone that for a followup branch.